include/network_interface/network_utils.h

```diff
@@ -83,14 +87,17 @@ namespace Network
 							const float& factor,
 							const unsigned int& valueOffset)
 	{
-		unsigned long rcvData = 0;
+		uint64_t rcvData = 0;
 
 		for (unsigned int i = 0; i <  size; i++) {
 			rcvData <<= 8;
 			rcvData |= bufArray[(offset) + i];
 		}
-
-		T retVal = (*(reinterpret_cast<T *>(&rcvData)) * (T)factor) - valueOffset;
+		
+		T retVal;
+		std::memcpy(&retVal, &rcvData, sizeof(T));
```

Comment:
The problem with using std::memcpy here is machine endianness. Whether the data gets copied from the left or right side of rcvData depends on the endianness of the machine that is running the program and since we are using bitshifts above to store the data, the underlying memory storage of said data could be different depending on the machine's endianness. The safest and most portable way to do this is to shift the value back out of rcvData. However, since we would be shifting it into rcvData and right back out again, why don't we just drop rcvData and shift it straight into a retVal of type T? As long as (sizeof(T) == size) ( which should probably be checked for) then the endianness becomes irrelevant.

Reply:
So, one possibility is to handle floats specially. If sizeof(T) == sizeof(float), then memcpy it to a uint32_t. If sizeof(T) == sizeof(double), then memcpy it to a uint64_t. We probably won't need to handle the case of sizeof(long double) (128 bits).

Reply:
In this case, memcpy is OK because the underlying storage order will be the same for the float and the unsigned integer and we aren't copying a subset.